client/tailscale/localclient.go

```diff
@@ -1124,19 +1125,81 @@ func (lc *LocalClient) DeleteProfile(ctx context.Context, profile ipn.ProfileID)
 	return err
 }
 
-// QueryFeature makes a request for instructions on how to enable a
-// feature, such as Funnel, for the node's tailnet.
+// EnableFeatureInteractive sends the node's user through an interactive
+// flow to enable a tailnet feature, such as Funnel.
 //
-// This request itself does not directly enable the feature on behalf
-// of the node, but rather returns information that can be presented
-// to the acting user about where/how to enable the feature.
+// The hasRequiredCapabilities parameter should be a func that checks
+// whether a slice of node capabilities encloses the necessary values
+// needed to use the feature.
 //
-// If relevant, this includes a control URL the user can visit to
-// explicitly consent to using the feature. LocalClient.WatchIPNBus
-// can be used to block on the feature being enabled.
+// If ok is returned true, the feature has been successfully enabled.
 //
-// 2023-08-02: Valid feature values are "serve" and "funnel".
-func (lc *LocalClient) QueryFeature(ctx context.Context, feature string) (*tailcfg.QueryFeatureResponse, error) {
+// If err is returned non-empty, the client failed to query the control
+// server for information about how to enable the feature.
+//
+// If ok is false and err is nil, the feature could not be enabled
+// interactively (maybe this user doesn't have permission to enable it,
+// or they're on the wrong billing plan).
+// In this case, the CLI should exit.
+//
+// 2023-08-03: Valid feature values are "serve" and "funnel".
+func (lc *LocalClient) EnableFeatureInteractive(ctx context.Context, feature string, hasRequiredCapabilities func(caps []string) bool) (ok bool, err error) {
```

Comment:
This seems pretty hyper-specialized for being in the tailscale client package where we kinda have an API promise. Plus that func callback is pretty specific.

This seems like it could live elsewhere, like in the CLI package even? Does anything else need to call this besides the CLI?

Reply:
Okay, I moved this to a new `interactive.go` file in the `cli` package. Was gonna just put it in `serve.go` but then we'll probably end up moving it in the future (there's been some chatter about using the interactive consent flow for SSH too). Happy to move it to `serve.go` for now though if you've got strong opinions on that!

Reply:
Actually, I ended up changing my mind and keeping it in `serve.go` for now to keep things simple. Take a quick look whenever you get a chance.